Google.PowerShell/BigQuery/BqTabledata.cs

```diff
@@ -230,4 +231,206 @@ protected override void EndProcessing()
+
+        /// <summary>
+        /// <para type="description">
+        /// The format of the data file (Ex: CSV, JSON).
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, Position = 1)]
+        public DataFormats Type { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// Write Disposition of the operation. Governs what happens to the data currently in the table.
+        /// If this parameter is not supplied, this defaults to WriteAppend
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        public WriteDisposition WriteMode { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The number of malformed rows that the request will ignore before throwing an error. 
+        /// This value is zero if not specified.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        public int MaxBadRecords { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// CSV ONLY: Allows insertion of rows with fields that are not in the schema, ignoring the extra fields.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        public SwitchParameter AllowUnknownFields { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// CSV ONLY: Allows insertion of rows that are missing trailing optional columns.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        public SwitchParameter AllowJaggedRows { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// CSV ONLY: Allows quoted data sections to contain newlines
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        public SwitchParameter AllowQuotedNewlines { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// CSV ONLY: Separator between fields in the data.  Default value is comma (,).
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        public string FieldDelimiter { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// CSV ONLY: Value used to quote data sections.  Default value is double-quote (").
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        public string Quote { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// CSV ONLY: The number of rows to skip from the input file. (Usually used for headers.)
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        public int SkipLeadingRows { get; set; }
+
+        protected override void ProcessRecord()
+        {
+            Stream fileInput;
+            try
+            {
+                fileInput = new StreamReader(Filename).BaseStream;
```

Comment:
Streams should be controlled by a using statment. A stream should not be left unclosed.

```
using (Stream fileInput = File.OpenRead(Filename)) {
    ....
}
```